Air_Traffic_Control_System/src/ResourceManager.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * ResourceManager.cpp
+ *
+ *  Created on: Mar 14, 2023
+ *      Author: Argi
+ */
+
+#include "ResourceManager.h"
+
+ResourceManager::ResourceManager() {
+	// TODO Auto-generated constructor stub
+
+}
+
+ResourceManager::~ResourceManager() {
+	// TODO Auto-generated destructor stub
+}
+
+
+int ResourceManager::spawnNewAircraftThread(Aircraft AircraftObj){
+
+	for (i=1;i<=numOfLines;i++)
+	{
+		int pid=i;
+		pthread_create(NULL, NULL, &functionA ,NULL);
+
+		pthread_create( NULL, NULL, &function, NULL );
+	}
```

Comment:
Keep it as a placeholder, more research needed to figure out the proper way to create threads and execute them in QNX. More importantly, in the resourceManager start with the takeNextAircraft function, however we need the Parser to be fully operational for that. I would say for now just try implementing the SSR since you know the type of data you will be receiving.